node/src/main/scala/coop/rchain/node/Main.scala

```diff
@@ -106,18 +166,68 @@ object Main {
     )
   }
 
-  private def generateKey(conf: Configuration, algorithm: String): Task[Unit] =
+  private def generateKey(
+      conf: Configuration,
+      algorithm: String,
+      privateKeyPath: Path
+  ): Task[Unit] =
     for {
-      keyPair <- algorithm.toLowerCase match {
-                  case Ed25519.name   => Ed25519.newKeyPair.pure[Task]
-                  case Secp256k1.name => Secp256k1.newKeyPair.pure[Task]
-                  case _              => Task.raiseError(new IllegalStateException("Invalid algorithm name"))
-                }
-      (sec, pub) = keyPair
-      sk         = Base16.encode(sec.bytes)
-      pk         = Base16.encode(pub.bytes)
-      _ <- ConsoleIO[Task].println(s"Generated public key: $pk") >>
-            ConsoleIO[Task].println(s"Generated private key: $sk")
+      sk <- algorithm.toLowerCase match {
+             case Ed25519.name =>
+               val provider   = new BouncyCastleProvider()
+               val (sk, _)    = Ed25519.newKeyPair
+               val keyFactory = KeyFactory.getInstance("Ed25519", provider)
+               val privKeyInfo = new PrivateKeyInfo(
+                 new AlgorithmIdentifier(EdECObjectIdentifiers.id_Ed25519),
+                 new DEROctetString(sk.bytes)
+               )
+               val pkcs8KeySpec = new PKCS8EncodedKeySpec(privKeyInfo.getEncoded)
+               keyFactory.generatePrivate(pkcs8KeySpec).pure[Task]
+             case Secp256k1.name =>
+               val provider         = new BouncyCastleProvider()
+               val keyPairGenerator = KeyPairGenerator.getInstance("ECDSA", provider)
+               keyPairGenerator.initialize(
+                 new ECGenParameterSpec("secp256k1"),
+                 SecureRandomUtil.secureRandomNonBlocking
+               )
+               val kp = keyPairGenerator.generateKeyPair
+               kp.getPrivate.pure[Task]
+             case _ => Task.raiseError(new IllegalStateException("Invalid algorithm name"))
+           }
+      password       <- ConsoleIO[Task].readPassword("Password for generated private key file: ")
+      passwordRepeat <- ConsoleIO[Task].readPassword("Repeat password: ")
+      _ <- if (password != passwordRepeat) {
+            ConsoleIO[Task].println("Passwords do not match. Try again.") >> generateKey(
```

Comment:
you don't need to re-generate the key here. You only have to re-read the password